src/main/java/com/apeshko/javaspring/dao/Dao.java

```diff
@@ -0,0 +1,12 @@
+package com.apeshko.javaspring.dao;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface Dao<T> {
```

Comment:
that's fine, just recommendation, like dao implemented in Spring Data JPA, generic class may be expanded a bit like this (to make it more flexible):
```
public interface Repository<T,ID> {
  T getById(ID id);
    List<T> getAll();
    T save (T t);
    T update(T t);
    boolean delete(ID id);
}
```